### PR TITLE
Separate websupport module

### PR DESCRIPTION
**Pull request by tk0miya**
As I proposed in #3254, I try to separate websupport module to sphinxcontrib-websupport (see https://github.com/sphinx-doc/sphinxcontrib-websupport).

* [x] CHANGES
* [x] docs
* [x] Update tests (to be slim and quick)
* [x] emit deprecation warnings

### CODE REVIEW

**Comment by tk0miya** [comment]
@shimizukawa Could you try this with your apps?
I need to ensure this is fine by your confirmation.

**Comment by tk0miya** [comment]
Finally, I finished all my work.
Please review this please.
(https://github.com/sphinx-doc/sphinxcontrib-websupport package also!)

File: tox.ini [hunk 1]
```diff
@@ -5,12 +5,11 @@ envlist=flake8,py27,py34,py35,py36,pypy,du13,du12,du11
 deps=
     six
     pytest
-    sqlalchemy
-    whoosh
     html5lib
     mock
     enum34
     typing
+    https://github.com/sphinx-doc/sphinxcontrib-websupport/archive/master.zip
```
Comment (tk0miya) [inline tox.ini]: I will remove this before merging.

**Review by shimizukawa — COMMENTED** [review]
LGTM!

I tested with https://github.com/shimizukawa/sphinx-websupport-app/tree/1fae41ae16301e506bdf666ca2baf8a69e78acaa

**Review by shimizukawa — APPROVED** [review]

**Comment by shimizukawa** [comment]
image

**Comment by tk0miya** [comment]
Merged!

**Pull request merged**